### DISCUSSION TITLE
Middleware just on `post` route

### DISCUSSION THREAD

**Discussion by matheusrocha89** [vercel/next.js#14933] [Help] [answered]
Hi people,
I have a situation where I created an authorization middleware for my API. The situation is that I use the middleware with `nextConnect` and then I do:

```javascript
const handler = nextConnect();

handler.use(databaseMiddleware);

handler.get(.........);
handler.post(.......);

export default handler
```

I want to use my authorization middleware just for the `post` route and not for the get. But if I do `handler.use()` it will be used by both, is there a way to do that?

**Comment by musme** [comment, marked as answer]
> Hi people,
> I have a situation where I created an authorization middleware for my API. The situation is that I use the middleware with `nextConnect` and then I do:
> 
> ```js
> const handler = nextConnect();
> 
> handler.use(databaseMiddleware);
> 
> handler.get(.........);
> handler.post(.......);
> 
> export default handler
> ```
> 
> I want to use my authorization middleware just for the `post` route and not for the get. But if I do `handler.use()` it will be used by both, is there a way to do that?

Hi! What about this?

`handler.post('/', databaseMiddleware);`

**Reply by matheusrocha89** [reply]
Thanks @musme I will try that.

I should have thought that but I was very busy and the brain was tired 😄

**Reply by matheusrocha89** [reply]
When I do that he can't find the route, give me 404.

**Reply by musme** [reply]
Can you try without '/' ?

`handler.post(databaseMiddleware, yourPostHandle);`

**Reply by matheusrocha89** [reply]
Same thing, 404. The only solution I found was doing this:
```javascript
import nextConnect from 'next-connect';

import {
  firebaseMiddleware,
  authorizationMiddleware,
} from '../../../middlewares';

const handler = nextConnect();
const commonMiddlewares = nextConnect();
const handlerWithAuth = nextConnect();

commonMiddlewares.use(firebaseMiddleware);
handlerWithAuth.use(authorizationMiddleware);

handlerWithAuth.post(async (req, res) => { ... });

handler
  .use(commonMiddlewares)
  .post(authHandler);

export default handler;
```

**Reply by matheusrocha89** [reply]
Found the issue, your way is correct @musme 
Our middleware was created in a wrong way, that's why your first solution was giving me 404, but now it's working perfectly the way you said. Thanks.
